data-serving/data-service/src/model/source.ts

```diff
@@ -1,22 +1,13 @@
 import mongoose from 'mongoose';
 
 const fieldRequiredValidator = [
-    function (this: SourceDocument) {
-        return (
-            this != null &&
-            this.id == null &&
-            this.url == null &&
-            this.other == null
-        );
+    function (this: SourceDocument): boolean {
+        return this != null && this.url == null && this.other == null;
     },
     'Source must specify id, url, or other',
 ];
 
 export const sourceSchema = new mongoose.Schema({
-    id: {
-        type: String,
-        required: fieldRequiredValidator,
-    },
```

Comment:
ooc, why not store the source id?

Reply:
_id/id is automatically added by mongoose to all documents